src/org/opensolaris/opengrok/index/IndexDatabase.java

```diff
@@ -395,6 +397,13 @@ public void update(IndexerParallelizer parallelizer)
             IndexWriterConfig iwc = new IndexWriterConfig(analyzer);
             iwc.setOpenMode(OpenMode.CREATE_OR_APPEND);
             iwc.setRAMBufferSizeMB(env.getRamBufferSize());
+            /**
+             * Most data in OpenGrok is indexed but not stored, so use the best
+             * compression on the minority of data that is stored, since it
+             * should not have a detrimental impact on overall throughput.
+             */
+            iwc.setCodec(new Lucene70Codec(
+                Lucene50StoredFieldsFormat.Mode.BEST_COMPRESSION));
```

Comment:
lucene usually has some "latest" links pointing to latest available codecs and formats, does it have it for those two too?
the idea I have is that upon every lucene upgrade everything (possible) should get upgraded automatically to latest greatest
(I know, downside is that you need to index from scratch and all the bugs get automatically picked up, but engine upgrade is transparent)

Reply:
ok, we shall stick with this exact matching then, if I will find a way around I will change it to generic to be able to stick to latest lucene (yes, it breaks backwards compatibility, but this is a developer tool, latest greatest is of the essence)